addons/stock_landed_costs/models/stock_landed_costs.py

```diff
@@ -0,0 +1,283 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import product
+from openerp import _, api, fields, models
+from openerp.tools import float_compare, float_round
+from openerp.exceptions import UserError
+import openerp.addons.decimal_precision as dp
+
+
+class StockLandedCost(models.Model):
+    _name = 'stock.landed.cost'
+    _description = 'Stock Landed Cost'
+    _inherit = 'mail.thread'
+
+    def get_valuation_lines(self, picking_ids):
+        lines = []
+        if not picking_ids:
+            return lines
+
+        for picking in self.env['stock.picking'].browse(picking_ids):
+            for move in picking.move_lines:
+                #it doesn't make sense to make a landed cost for a product that isn't set as being valuated in real time at real cost
+                if move.product_id.valuation != 'real_time' or move.product_id.cost_method != 'real':
+                    continue
+                total_cost = 0.0
+                weight = move.product_id and move.product_id.weight * move.product_qty
+                volume = move.product_id and move.product_id.volume * move.product_qty
+                for quant in move.quant_ids:
+                    total_cost += quant.cost * quant.qty
+                vals = dict(product_id=move.product_id.id, move_id=move.id, quantity=move.product_uom_qty, former_cost=total_cost, weight=weight, volume=volume)
+                lines.append(vals)
+        if not lines:
+            raise UserError(_('The selected picking does not contain any move that would be impacted by landed costs. Landed costs are only possible for products configured in real time valuation with real price costing method. Please make sure it is the case, or you selected the correct picking'))
+        return lines
+
+    name = fields.Char(track_visibility='always', string="Name", default=lambda self: self.env['ir.sequence'].next_by_code('stock.landed.cost'), readonly=True, copy=False)
+    date = fields.Date(required=True, string="Date", states={'done': [('readonly', True)]}, default=fields.Date.context_today, track_visibility='onchange', copy=False)
```

Comment:
If variable name and string name should be same then no need to mention as argument because when you not give string framework should be take automatically variable name as a string with Camelcase.